linkerd/app/outbound/src/lib.rs

```diff
@@ -349,39 +286,23 @@ impl Config {
                     .push_map_target(endpoint::route)
                     .into_inner(),
             ))
-            .check_new_service::<endpoint::Profile, http::Request<_>>()
-            .push_map_target(endpoint::Profile::from)
-            // Discovers the service profile from the control plane and passes
-            // it to inner stack to build the router and traffic split.
-            .push(profiles::discover::layer(
-                profiles_client,
-                AllowProfile(allow_discovery),
-            ))
             .check_new_service::<HttpLogical, http::Request<_>>()
-            .push_on_response(svc::layers().box_http_response())
-            .cache(
-                svc::layers().push_on_response(
-                    svc::layers()
-                        .push_failfast(dispatch_timeout)
-                        .push_spawn_buffer_with_idle_timeout(buffer_capacity, cache_max_idle_age)
-                        .push(metrics.stack.layer(stack_labels("logical"))),
-                ),
-            )
-            .instrument(|logical: &HttpLogical| info_span!("logical", dst = %logical.dst))
-            .into_make_service()
-            .spawn_buffer(buffer_capacity)
-            .check_make_service::<HttpLogical, http::Request<_>>()
             .push(http::header_from_target::layer(CANONICAL_DST_HEADER))
-            // Strips headers that may be set by this proxy.
-            .push_on_response(http::strip_header::request::layer(DST_OVERRIDE_HEADER))
-            .check_make_service_clone::<HttpLogical, http::Request<B>>()
+            .push_on_response(
+                svc::layers()
+                    // Strips headers that may be set by this proxy.
+                    .push(http::strip_header::request::layer(DST_OVERRIDE_HEADER))
+                    .push(svc::layers().box_http_response()),
+            )
```

Comment:
since we no longer honor `DST_OVERRIDE`, does this still need to be here at all? or can we just stop caring about it?

Reply:
We should continue to strip the header through at least the stable release so that we're not all-of-a-sudden passing through the header if it's set by config.